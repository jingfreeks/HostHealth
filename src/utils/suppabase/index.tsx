export {default as supabase} from './suppabase';
