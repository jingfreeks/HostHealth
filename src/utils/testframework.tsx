import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
