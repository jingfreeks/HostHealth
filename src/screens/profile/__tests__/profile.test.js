import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import {renderWithProviders} from '@/utils/testframeworknew';

import Profile from '../profile';

jest.mock('react-native-image-picker', () => ({
  launchImageLibrary:jest.fn(),
}));
describe('Profile Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Profile />);
    expect(all.toJSON()).toMatchSnapshot();
  });

  it('Should work as expected to trigger logout button', () => {
    const all = renderWithProviders(<Profile />);
    const el=all.getByTestId('ProfileLogoutButton')
    fireEvent(el, 'onPress');
    expect(all.toJSON()).toBeTruthy();
  });

});
