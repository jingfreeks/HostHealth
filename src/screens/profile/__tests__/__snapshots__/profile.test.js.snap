// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "marginLeft": 20,
                },
              ]
            }
          >
            <View
              style={
                {
                  "backgroundColor": "transparent",
                  "borderRadius": 65,
                  "height": 130,
                  "width": 130,
                }
              }
              theme={
                {
                  "colors": {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": {
                      "android": {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 65,
                    "flex": 1,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <Image
                  borderRadius={65}
                  onLoad={[Function]}
                  source={
                    {
                      "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
                    }
                  }
                  style={
                    {
                      "bottom": 0,
                      "flex": 1,
                      "height": undefined,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image"
                  theme={
                    {
                      "colors": {
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": {
                          "android": {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                        "white": "#ffffff",
                      },
                    }
                  }
                  transition={true}
                  transitionDuration={360}
                />
                <View
                  accessibilityElementsHidden={true}
                  collapsable={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#bdbdbd",
                        "flex": 1,
                        "height": undefined,
                        "justifyContent": "center",
                        "width": undefined,
                      }
                    }
                    testID="RNE__Image__placeholder"
                  />
                </View>
                <View
                  style={
                    {
                      "flex": 1,
                      "height": undefined,
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image__children__container"
                />
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 35.311764705882354,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              undefined undefined undefined
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 35.311764705882354,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              Employee
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 35.311764705882354,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              Details
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 35.311764705882354,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              Bank info
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 35.311764705882354,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 0.1,
          "flexShrink": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      accessibilityLabel="ProfileLogoutButton"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
      testID="ProfileLogoutButton"
    >
      <Text
        style={
          [
            {
              "color": "#0e3f6c",
              "fontFamily": "Avenir",
              "fontSize": 35.311764705882354,
              "fontWeight": "900",
            },
          ]
        }
      >
        Log-out
      </Text>
    </View>
  </View>
</View>
`;
